ocaml/xapi/xapi_pif.ml

```diff
@@ -776,7 +776,7 @@ let pif_has_clustering_enabled ~__context (self : API.ref_PIF) network =
 
 let set_disallow_unplug ~__context ~self ~value =
   let network = Db.PIF.get_network ~__context ~self in
-  if pif_has_clustering_enabled ~__context self network
+  if ((Db.PIF.get_disallow_unplug ~__context ~self) <> value)&& pif_has_clustering_enabled ~__context self network
   then raise Api_errors.(Server_error(clustering_enabled_on_network, [Ref.string_of network]))
   else Db.PIF.set_disallow_unplug ~__context ~self ~value
```

Comment:
You could skip this if you found it to be equal, but its fine either way.

Reply:
Yeah, that's a better way to do it.